command/agent/srv.go

```diff
@@ -0,0 +1,120 @@
+package agent
+
+import (
+	"fmt"
+	"io"
+	"log"
+	"net"
+	"time"
+
+	"github.com/hashicorp/serf/serf"
+)
+
+const (
+	srvPollInterval = 60 * time.Second
+)
+
+// AgentSRV periodically polls an SRV record
+// And attempts to join the hosts supplied
+type AgentSRV struct {
+	agent      *Agent
+	srvrecords []string
+	logger     *log.Logger
+	replay     bool
+	retry      bool
+}
+
+// NewAgentSRV is used to create a new AgentSRV
+func NewAgentSRV(agent *Agent, logOutput io.Writer, replay bool, srvrecords []string, retry bool) (*AgentSRV, error) {
+
+	// Initialize the AgentSRV
+	m := &AgentSRV{
+		agent:      agent,
+		srvrecords: srvrecords,
+		logger:     log.New(logOutput, "", log.LstdFlags),
+		replay:     replay,
+		retry:      retry,
+	}
+
+	if m.retry {
+		// Start the poller in the background
+		m.logger.Printf("[INFO] Starting SRV background poller.")
+		go m.poll()
+	} else {
+		m.logger.Printf("[INFO] Starting SRV for one-shot join.")
+		// A one-shot attempt to try to join the cluster via SRV
+		go m.joinSRV()
+	}
+	return m, nil
+}
+
+// run is a long running goroutine that scans for new hosts periodically
+func (m *AgentSRV) poll() {
```

Comment:
Kind of related to above, I think even this one stops once it has been able to join something.